crates/ty_python_semantic/resources/mdtest/generics/pep695/variance.md

```diff
@@ -380,4 +336,296 @@ static_assert(not is_equivalent_to(D[Any], C[Any]))
 static_assert(not is_equivalent_to(D[Any], C[Unknown]))
 ```
 
+## Mutual Recursion
+
+This example due to Martin Huschenbett's PyCon 2025 talk,
+[Linear Time variance Inference for PEP 695][linear-time-variance-talk]
+
+```py
+from ty_extensions import is_subtype_of, static_assert
+from typing import Any
+
+class A: ...
+class B(A): ...
+
+class C[X]:
+    def f(self) -> "D[X]":
+        return D()
+
+    def g(self, x: X) -> None: ...
+
+class D[Y]:
+    def h(self) -> C[Y]:
+        return C()
+```
+
+`X` is contravariant in both `C` and `D`:
+
+- `C` has two occurrences of `X`
+    - `X` occurs in the return type of `f` as `D[X]` (`X` is substituted in for `Y`)
+        - `D` has one occurrence of `Y`
+            - `Y` occurs in the return type of `h` as `C[Y]`
+    - `X` occurs contravariantly as a parameter in `g`
+
+We want the _greatest_ inferrable variance, so we start by assuming `X` occurs bivariantly in `C`.
+Therefore, `X` has one bivariant occurrence, and one contravariant occurrence, which resolves to
+contravariant.
+
+Recall that `Y`'s only occurrence in `D` is covariantly in `C[Y]` (`Y` subbing in for `X`). We just
+stated that `X` is contravariant in `C`, so we conclude that `Y` is contravariant in `D`.
+
+```py
+static_assert(not is_subtype_of(C[B], C[A]))
+static_assert(is_subtype_of(C[A], C[B]))
+static_assert(not is_subtype_of(C[A], C[Any]))
+static_assert(not is_subtype_of(C[B], C[Any]))
+static_assert(not is_subtype_of(C[Any], C[A]))
+static_assert(not is_subtype_of(C[Any], C[B]))
+
+static_assert(not is_subtype_of(D[B], D[A]))
+static_assert(is_subtype_of(D[A], D[B]))
+static_assert(not is_subtype_of(D[A], D[Any]))
+static_assert(not is_subtype_of(D[B], D[Any]))
+static_assert(not is_subtype_of(D[Any], D[A]))
+static_assert(not is_subtype_of(D[Any], D[B]))
+```
+
+## Class Attributes
+
+### Mutable Attributes
+
+Normal attribtues are mutable, and so are invariant (see [inv]).
```

Comment:
```suggestion
Normal attributes are mutable, and so are invariant (see [inv]).
```

Reply:
Perhaps pedantically, I don't think you can say that the attribute itself is invariant; rather, their presence leads us to infer that the enclosing generic class in invariant over this TypeVar.